### DISCUSSION TITLE
Looking for hardware suggestions

### DISCUSSION THREAD

**Discussion by GPappy1959** [collin80/SavvyCAN#585] [General]
I was planning on using a Macchina m2 but it looks like they are not available right now.  Looking for recommendations on a good hardware setup (and where it is available) to use with SavvyCAN to start sniffing.  I have a couple of projects that I would like to do and need to get more familiar with the Can Bus messages in my truck.

**Comment by collin80** [comment]
Unfortunately the chip shortage hit a lot of vendors very hard. The underlying chip used on the M2 has been more or less discontinued. I think the vendor claims that stock may eventually exist but so far there has been nothing. EVTV still has a variety of hardware such as this: https://store.evtv.me/products/esp32

Alternatively, you can use a PeakCAN adapter, a KVaser Leaf Light, anything socketcan compatible (if you can run linux). Or, if you're adventurous, a lot of people have built their own ESP32 based CAN capture device from other parts available on the internet. I publish the source code for ESP32RET so it could be adapted to basically any ESP32 hardware you can get but you're on your own if you don't use the EVTV hardware. Or, at least I won't be able to help you. Other people may be able to offer some assistance in getting custom hardware to work.

**Reply by GPappy1959** [reply]
Collin,
Thank you for taking the time to reply.  I had done some more poking around after my post and while there are a lot of options, none seemed to be what I was hoping for.  Since you wrote the code I guess you would be the expert so I ordered an EVTV CANDUE board and will give it a try. Bought the ESP32 book as well. Thanks.

**Reply by JoeBargo** [reply]
If you only need 1 bus, get a meatpi WiCAN. It is the easiest to get started with and it's great value. I have the EVTV ESP32 CANdue with Collin's name on it too. The CANdue board has more potential but it might take a little work to get it to do what you want. 

Also, I'm not sure how meatpi's stock is since they just released them. I ordered my wican-usb in November and just got it a week or two ago... EVTV's product was at my house in 2-3 days.

**Reply by GPappy1959** [reply]
Thank you for the thoughts Joe.

I just decided to go with Collin's suggestion and hope that he gets some profit off of it.  Worth paying a bit more to get the support and hopefully support the person who wrote the code.  It will be used for CAN Bus sniffing and some microcontroller projects I have been considering.  I can always purchase lower cost ESP32 boards for implementation once I am done.  I am really looking forward to getting the EVTV board to start playing around.

Greg

**Reply by spot2000** [reply]
@JoeBargo any advice or links into finding info on how to make use of meatpci WiCAN on SavyCAN?

**Reply by guhlandreas** [reply]
Hi all,
I would also be happy to get some help with respect to my WiCan and SavvyCAN.
In particular I do not receive any data and I am wondering why. I configured the WiCan to use the SavvyCAN protocol. But not sure what to do else ....
Thanks for your help!
Andreas

**Reply by inferiorhumanorgans** [reply]
> Unfortunately the chip shortage hit a lot of vendors very hard. The underlying chip used on the M2 has been more or less discontinued.

https://www.microchip.com/en-us/product/ATSAM3X8E

As of now (June 2024) Microchip is showing about 500 surface mount and 40,000 BGA in stock, with more expected in Aug/Sep.  Digikey is showing about 1,200 and 200 in stock respectively.  Mouser is showing 6,500 an 3,500 respectively.

Certainly I hope this isn't the end of the line for the 3x8e or the M2.

**Reply by collin80** [reply]
> As of now (June 2024) Microchip is showing about 500 surface mount and 40,000 BGA in stock, with more expected in Aug/Sep. Digikey is showing about 1,200 and 200 in stock respectively. Mouser is showing 6,500 an 3,500 respectively.
> 
> Certainly I hope this isn't the end of the line for the 3x8e or the M2.

Yes, despite all odds, they did start to produce the chips again! This surprised me but it's a good surprise. They aren't making very many of them but some is better than none. Well, as you show, BGA seems to be in much greater production but everyone I know was using surface mount so that's the more important one for hobbyist use. Anyway, EVTV is producing new CANDue boards with these chips. However, the bad news is that I don't think Macchina is making any more boards. So far as I can tell, they have more or less had to move on to other things since the chip shortage really put a dent into their ability to sell products. I will reach out to them to ask about the future though.

But, I do believe Arduino is making new Due boards so that will also be an option for people. All in all, it seems SAM3X has risen from the dead and is available in small quantities once again. However, the price has gone up so everything is likely to raise in price a bit.

**Reply by inferiorhumanorgans** [reply]
Yeah, unfortunately I'm not interested in CAN at all so the M2 fills a specific niche otherwise the A0 would've been enough.  That said I've an M2 already but it'd be nice for someone else with a pre-CAN car to be able to pick up an M2.

**Comment by EngMarc** [comment]
GP and Joe and Collin,
Thanks for the project and info.
I'm a kind of a newbie (not so much to Canbus but more to reading and sending data on it). I've done a lot of work with my Toyota Prius back a few years ago. 

Now, I have a custom project I'm working on to decipher the Canbus data traffic and hopefully build some logic and commands to use the bus to do some things useful to many of us. 

I've ordered CSS crocodile and CL2000 CAN Bus Logger & USB Interface and am going to SavvyCan to start sniffing on the canbus of my generator. It is built in Italy by Dometic and has zero support but I do have a wiring diagram so know where to tap in on the Can High and Can Low. That's all I have though at this point. 

What I'm looking for in the future is similar to some of the suggestions here which is what can I use to easily send commands on the CanBus both in prototype and in production (e.g. something low cost with firmware on it that says A -> canbus command A, B -> canbus command B?  That part is new for me since I've only read/listened to canbus traffic in the past. 

So, anyone out there that has built a canbus command sending unit, send them my way if they are willing to help a newbie a little bit.

**Comment by srolf** [comment]
At least for sniffing under Linux, a [USB2CAN from innomaker](https://www.inno-maker.com/product/usb2can-cable/) would do the job (as long as no CAN FD is required, but while speaking about DUE and ESP32 this shouldn't be a limitation). I own one of them and they are on Ubuntu 22.04 LTS 64bit recognized on x86 and Raspberry Pi 4 with the stock kernel (socketCAN).

If a programmable micro-controller is needed, maybe it's even an option to have a look at the new Arduino [UNO R4 Minima](https://store.arduino.cc/products/uno-r4-minima) (not the same as UNO R3 at all). The Renesas chip has got an integrated CAN controller with one channel.  I started some first steps by combining it with a TJA1051 transceiver. At the beginning there was a small hickup about SFF/EFF, but this is meanwhile by its Arduino [core release 1.04](https://github.com/arduino/ArduinoCore-renesas) resolved.

**Reply by anonymouselojz** [reply]
Hi Srolf, We released the SavvyCAN Software to support InnoMaker CAN And CANFD [Products.](https://github.com/INNO-MAKER/usb2can/tree/master/For%20Windows/SavvyCAN-InnoMaker)
![微信图片_20231225162432](https://github.com/collin80/SavvyCAN/assets/97109297/b414a1f2-b33b-4822-8102-93966f54fe10)
![微信图片_20231225222029](https://github.com/collin80/SavvyCAN/assets/97109297/2bf922e8-c7bd-429a-aada-136132d10e90)

**Comment by johnyhack** [comment]
If you would like to buy a Macchina M2, I have 4 new new ones available
